### PR TITLE
Integrating validator methods into config

### PR DESCRIPTION
**Pull request by rubenvecg**
Part of #129. I made a partial version of validator to return true on all checks. Called validator in `config.compute()`. If `validator()` returns true, compute proceeds to parse the unrendered file. Otherwise, compute returns `None`.

The new version of validator will take an already parsed yml file

### CODE REVIEW

File: library/config.py [hunk 1]
```diff
@@ -78,6 +79,11 @@ def version_today(self) -> str:
     @property
     def compute(self) -> dict:
         """based on given yml file, compute the configuration"""
+
+        # Validate unparsed, unrendered file
+        v = Validator(self.parsed_unrendered_template)
+        v()
```
Comment (SPTKL) [inline library/config.py:85]: can we call it validator? instead of v, we want to be verbose and specific

Comment (SPTKL) [reply library/config.py:85]: `Validator(self.parsed_unrendered_template)()` you can also do this

File: library/validator.py [hunk 2]
```diff
@@ -73,72 +73,28 @@ class Validator:
...
-        return (
-            (("socrata" not in source_fields) and ("script" not in source_fields))
-            if ("url" in source_fields)
-            else (("socrata" in source_fields) ^ ("script" in source_fields))
-        )
+        return True
 
+    # Assert all the individual checks
     @property
     def file_is_valid(self):
```
Comment (SPTKL) [inline library/validator.py:99]: I think this should be the call method no? it's the method that ties everything together

File: library/config.py (hunk 1 above)
Comment (rubenvecg) [reply library/config.py:85]: I agree, will change

File: library/validator.py (hunk 2 above)
Comment (rubenvecg) [reply library/validator.py:99]: I thought of that distinction for readability, or would it be redundant?

Comment (SPTKL) [reply library/validator.py:99]: it feels redundant, and the call method should be exactly what `file_is_valid` is

Comment (SPTKL) [reply library/validator.py:99]: but not as a property

File: library/validator.py [hunk 3]
```diff
@@ -73,72 +73,25 @@ class Validator:
...
-            return y
+    def __call__(self):
+        assert self.tree_is_valid
+        assert self.dataset_name_matches
+        assert self.has_only_one_source
 
+    # Check that the tree structure fits the specified schema
     @property
     def tree_is_valid(self) -> bool:
-        if self.__file["dataset"] == None:
```
Comment (SPTKL) [inline library/validator.py:98]: looking good! now you can bring back the deleted logic so that they are not returning the default True

**Review by SPTKL — APPROVED** [review]

**Pull request merged**